ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/ObservableBodyTest.kt

```diff
@@ -96,4 +98,114 @@ class ObservableBodyTest : ClientLoader() {
             }
         }
     }
+
+    @Test
+    fun testReceiveDataClassWithExecute() = clientTests {
+        config {
+            install(JsonFeature) {
+                serializer = KotlinxSerializer()
+            }
+        }
+
+        test { client ->
+            invokedCount = 0
+            val listener: ProgressListener = { _, _ -> invokedCount++ }
+
+            client.get<HttpStatement>("$TEST_SERVER/json/users-long") {
+                contentType(ContentType.Application.Json)
+                downloadProgress(listener)
```

Comment:
We also could think about a better name here: the current name doesn't highlight the intention
- `listenProgress { }`
- `onDownload`
- `setProgressListener` or `progressListener`

Reply:
i like `onDownload` to specify that it's the download and not upload progress

Reply:
btw we also may want to make something like this for uploading

Reply:
It's already merged #2450

Reply:
Updated this PR to unify API